src/thor/worker.cc

```diff
@@ -323,6 +323,39 @@ void thor_worker_t::log_admin(const valhalla::TripLeg& trip_path) {
   }
 }
 
+/*
+ * Apply attribute filters from the request to the AttributesController. These filters
+ * allow including or excluding specific attributes from the response. These filters
+ * are honored in route, trace_route, and trace_attributes actions.
+ */
+void thor_worker_t::filter_attributes(const valhalla_request_t& request,
```

Comment:
i wonder we should:

1. make AttributesController a member variable
2.turn this function into a `parse_attributes_controller` function and just call it at the top of each action like all the other `parse_*` functions are called


then we dont have to pass it all over the place. i feel like there might be a gotcha in here either way in the default. seems like when the request doesnt include a filter action we enable everything but this is probably not the same as the default instantiated controller. we need to reconciile this some how

Reply:
>  seems like when the request doesnt include a filter action we enable everything but this is probably not the same as the default instantiated controller

Yeah I think you're right that we didn't enable_all before, but looking at the [definition](https://github.com/valhalla/valhalla/blob/master/src/thor/attributes_controller.cc#L10-L109) it looks like it sets everything to true initially? In that case calling enable_all won't have any affect right now. However if/when we add more keys that we want to default to false then we definitely don't want to call enable_all by default. I think I can rearrange the logic here a bit to only enable_all for trace_attributes and keep the default behavior unchanged for the others.

Reply:
Another detail to note is that the include and exclude filters are "strict" meaning if you include a filter.attribute in your request we either include _everything except_ fields specified in `exclude` or we exclude _everything except_ fields specified in `include`. 

This is different than the behavior I wanted to implement which is more about modifying the default response fields: include means add the field to the response if it's not already included, and exclude means remove the field from the response whether it would've been included or not. 

I hope that makes some sense! I'll make some changes in this PR and run them by you for a gut check.